### PR TITLE
Update Ruby to version 2.4.3

### PR DESCRIPTION
**Pull request by ykzts**
- [Ruby 2.4.3 Released - Ruby Programming Language](https://www.ruby-lang.org/en/news/2017/12/14/ruby-2-4-3-released/)

> Ruby 2.4.3 has been released.
> 
> This release includes some bug fixes and a security fix.
> 
> - [CVE-2017-17405: Command injection vulnerability in Net::FTP](https://www.ruby-lang.org/en/news/2017/12/14/net-ftp-command-injection-cve-2017-17405/)
> 
> There are also som bug fixes. See [commit logs](https://github.com/ruby/ruby/compare/v2_4_2...v2_4_3) for more details.

- [x] rbenv
- [x] rvm
- [x] Travis
- [x] Docker (https://github.com/docker-library/official-images/pull/3810)

### CODE REVIEW

**Comment by ykzts** [comment]
https://github.com/docker-library/official-images/pull/3810 has been merged :tada:

**Comment by larouxn** [comment]
I'm pretty sure we should also run a `bundle install` locally so the `Gemfile.lock` is updated with Ruby 2.4.3 along with these other files.
https://github.com/tootsuite/mastodon/blob/a434d9c0cc89a0dd862c0f28825e1a1255e8a028/Gemfile.lock#L642-L643

**Comment by ykzts** [comment]
It is not necessary to update Gemfile.lock.

**Comment by larouxn** [comment]
@ykzts, out of curiosity, why is it unnecessary?

**Comment by Gargron** [comment]
@larouxn Versions of Ruby gems do not change regardless of Ruby version, so it would be a line change without any effects.

File: .travis.yml [hunk 1]
```diff
@@ -37,8 +37,8 @@ addons:
     - yarn
 
 rvm:
-  - 2.3.4
-  - 2.4.2
+  - 2.3.6
```
Comment (Gargron) [inline .travis.yml]: Does anyone think we still need to test on the 2.3 branch?

**Comment by larouxn** [comment]
@Gargron, ah, that's true. I guess it'll get updated next time we bump/add a gem.

File: .travis.yml (hunk 1 above)
Comment (ykzts) [reply .travis.yml]: Ruby 2.5 will be released on the 24th of this month. I want to add CI 2.5 and drop 2.3.

**Pull request closed without merging**